### ISSUE TITLE
Should configparser method signatures permit fallback=None?

### ISSUE THREAD

**Issue by ExplodingCabbage** [python/typeshed#2093]
The docs at https://docs.python.org/3/library/configparser.html explicitly give an example of using `fallback=None`:

```
print(cfg.get('Section1', 'monster', fallback=None))
```

This is not permitted with the current typeshed signatures.

**Comment by ExplodingCabbage** [comment]
I guess the awkward thing here is that changing this would require changing the *return type* of `get` to be `Optional[str]`, too... unless there's some clever way using generics or something to specifically hint that it can *only* return `None` in the case where the fallback is `None`?

**Comment by languitar** [comment]
Just to stress this, the doc says:

> None can be provided as a fallback value.

**Comment by wiml** [comment]
I've run into this as well: I think the correct type annotation is similar to the one used for `dict.get(k, default=v)`, where the default is given a type variable `T` and the return value has type `Union[T, int]`.  This is equivalent to `Optional[int]` if your fallback is None, but also allows other sentinel values to be used (and types correctly if the default is an int). 

https://github.com/python/typeshed/blob/2e0af18ddaa042644c3b2eb03f4f64ba7ef10c2f/stdlib/3/typing.pyi#L351

(I think an `@overload` is maybe still needed to handle the "raise if not found" behavior that you get if no fallback is specified.)

**Comment by The-Compiler** [comment]
In a somewhat similar vein, `SectionProxy` objects have a default fallback to `None`:

```python
import configparser

parser = configparser.ConfigParser()
parser.add_section('sect')

val1 = parser['sect'].get('key')
print(val1)  # None
reveal_type(val1)  # str
```

**Comment by The-Compiler** [comment]
I think this was fixed via 41435ef25a1054506c1ad6ea8c64bbe162944005 / #8380 as a fix for #7476?
